test_pull_requests

```diff
@@ -1650,7 +1805,12 @@ elsif mark_test_success_pull_id
 elsif mark_testonlyextended_success_pull_id
   @api_client.mark_test_success(mark_testonlyextended_success_pull_id, pull_id_repo, Properties['settings']['testonlyextended_settings'])
 elsif local_merge_pull_id
-  @api_client.local_merge_pull_request(local_merge_pull_id, pull_id_repo)
+  if ENV['MERGE'].to_s.downcase == 'true'
```

Comment:
We have to depend on `ENV['BUILD_URL']` in `base_commit_id_for_merge`, is that OK?

To get right of the `ENV['MERGE']` test I will need to adapt `extract_bot_comment_metadata` to do the right thing when `settings` is `nil`, but I think it's doable.